src/zarr/codecs/pipeline/hybrid.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import annotations
+
+from itertools import islice
+from typing import TYPE_CHECKING, TypeVar
+import numpy as np
+from dataclasses import dataclass
+
+from zarr.config import config
+from zarr.abc.codec import (
+    ByteGetter,
+    ByteSetter,
+    Codec,
+)
+from zarr.codecs.pipeline.batched import BatchedCodecPipeline
+from zarr.codecs.pipeline.core import CodecPipeline
+from zarr.common import concurrent_map
+
+if TYPE_CHECKING:
+    from typing import List, Optional, Tuple, Iterable
+    from typing_extensions import Self
+    from zarr.common import ArraySpec, BytesLike, SliceSelection
+
+T = TypeVar("T")
+
+
+def batched(iterable: Iterable[T], n: int) -> Iterable[Tuple[T, ...]]:
+    if n < 1:
+        raise ValueError("n must be at least one")
+    it = iter(iterable)
+    while batch := tuple(islice(it, n)):
+        yield batch
+
+
+@dataclass(frozen=True)
+class HybridCodecPipeline(CodecPipeline):
+    batch_size: int
+    batched_codec_pipeline: BatchedCodecPipeline
```

Comment:
Having a sibling class as an attribute is rather surprising. I see that this attribute is used quite extensively in this class, in combination with the `batched` function. This makes me wonder whether we need `HybridCodecPipeline` at all, as opposed to just putting a `batch_size` parameter on `BatchedCodecPipeline`

Reply:
I got rid of the `HybridCodecPipeline` and added a `batch_size` parameter to the `BatchedCodecPipeline`.